### DISCUSSION TITLE
is it right time to replace startwm.sh with standard distrib tool? (wm wont start but no error messages in the /var/log/xrdp-sesman.log)

### DISCUSSION THREAD

**Discussion by cypa** [neutrinolabs/xrdp#2260] [Ideas]
I see black square only after connect to RDP, but there is no error messages in /var/log/xrdp-sesman.log!
How do I debug what xrdp-sessman doing and what is the problem with startwm/xsession?

**Comment by cypa** [comment]
on CentOS 7 startx and lightdm work OK, but xrdp can't start wm - why?

**Comment by cypa** [comment]
finally problem was xterm wasn't installed so I've seen only block screen with cursor, but why mate-session wont start?

**Comment by cypa** [comment]
btw, I've found that we could replace startwm.sh with startx -- :11
so why use custom startwm.sh if we can use standard startx?

**Comment by matt335672** [comment]
`startx` attempts to start an X server I believe, so I don't think it's the right tool for the job here.

Following the code through, `xinit` will use `/etc/X11/xinit/xinitrc` to start the window manager (in the absence of any user config).

On  RHEL/CentOS/??? 7, this uses `/etc/X11/xinit/xinitrc-common`, which is also used by `/etc/X11/xinit/Xsession`. So the effects should be the same, more or less.

On other systems, `xinitrc-common` doesn't exist. This seems to be a Red Hat special, or they may use other mechanisms. On Debian/Ubuntu , `X11/xinit/xinitrc` just calls `/etc/X11/Xsession` directly. So `Xsession` is in a different place from RH. It's all a bit of a rats' nest. I don't think the 'standard distrib tool' you refer to above actually exists I'm afraid.

If you really want to use `startx`, and it works for you, you can replace startwm.sh with:-

```bash
#!/bin/sh

exec /usr/bin/startx -- "$DISPLAY" >~/.xsession-errors 2>&1
```

so you'll get a log file.

Personally, I'm not very happy with the existing startwm.sh approach as it's quite high maintenance, but I don't have a better solution at the moment. I'd like to modify sesman to remove the dependency on assuming the user is using bash, but that's not going to help hugely.

**Comment by matt335672** [comment]
I'm going to convert this old issue to a discussion. Maybe we'll get some more useful input that way.
